kafka/producer.go

```diff
@@ -173,8 +173,10 @@ func (p *Producer) gethandle() *handle {
 }
 
 func (p *Producer) produce(msg *Message, msgFlags int, deliveryChan chan Event) error {
-	if msg == nil || msg.TopicPartition.Topic == nil || len(*msg.TopicPartition.Topic) == 0 {
+	if msg == nil || len(*msg.TopicPartition.Topic) == 0 {
 		return newErrorFromString(ErrInvalidArg, "")
+	}else if msg.TopicPartition.Topic == nil{
+		return newErrorFromString(ErrUnknownTopic, "topic is null, please check")
```

Comment:
```suggestion
	} else if msg.TopicPartition.Topic == nil || len(*msg.TopicPartition.Topic) == 0 {
		return newErrorFromString(ErrInvalidArg, "topic cannot be nil or empty")
```

Reply:
We must keep in mind, 2 things, 
1. msg.TopicPartition.Topic == nil  must be made before len(*msg.TopicPartition.Topic) == 0 , else it causes a nil pointer dereference and will panic.
2. We can't change the type of the error, only the message of it (since the type is a part of the public API so people might be relying on that)